### ISSUE TITLE
`prefer-const` does not appear to detect destructuring assignments.

### ISSUE THREAD

**Issue by SimonSchick** [palantir/tslint#1892]
### Bug Report

- __TSLint version__: `4.1.1`
- __TypeScript version__: `2.0.10`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
let a;
    ~ Identifier 'a' is never reassigned; use 'const' instead of 'let'.
[a] = [1];
```

#### Actual behavior

Triggers a linting error.

#### Expected behavior

Should not violate the rule.
